src/main/java/org/terasology/taskSystem/TaskManagementSystem.java

```diff
@@ -49,94 +59,233 @@
             return;
         }
 
-        task.assignedTaskType = newTaskType;
+        taskComponent.assignedTaskType = newTaskType;
 
         if (newTaskType.equals(AssignedTaskType.Build)) {
-            task.buildingType = event.getBuildingType();
+            taskComponent.buildingType = event.getBuildingType();
         }
 
-        EntityRef taskEntity = entityManager.create(task);
+        //mark this area so that no other task can be assigned here
+        markArea(newBlockSelectionComponent, player);
+
+        taskEntity.addComponent(taskComponent);
+
+        addTask(player);
+    }
+
+    /**
+     * Saves the selected area for a particular task to check for clashes later.
+     * Attaches a {@link BlockSelectionComponent} to the assignedArea entity so that the assigned area remains colored
+     * until the task is finished.
+     * @param blockSelectionComponent
+     */
+    private void markArea(BlockSelectionComponent blockSelectionComponent, EntityRef player) {
+        AssignedAreaComponent assignedAreaComponent = new AssignedAreaComponent();
+
+        assignedAreaComponent.assignedRegion = blockSelectionComponent.currentSelection;
+
+        assignedAreaComponent.assignedTaskType = taskComponent.assignedTaskType;
+        assignedAreaComponent.buildingType = taskComponent.buildingType;
+
+        EntityRef assignedArea = entityManager.create(assignedAreaComponent, blockSelectionComponent);
+
+        oreonHolding.assignedAreas.add(assignedArea);
+
+        player.saveComponent(oreonHolding);
 
-        addTask(taskEntity, player);
+        logger.info("Adding new area to index : " + oreonHolding.assignedAreas.size() + " " + oreonHolding);
+        taskComponent.assignedAreaIndex = oreonHolding.assignedAreas.size() - 1;
+    }
+
+    /**
+     * Checks if the selected area can be used i.e not already assigned to some other task
+     * @param blockSelectionComponent
+     * @return A boolean value specifying whether the area is valid
+     */
+    private boolean checkArea(BlockSelectionComponent blockSelectionComponent) {
+        return true;
+    }
+
+    /**
+     * Looks for a building in the assignedAreas list.
+     * @param buildingType
+     * @return Returns a target for the Oreon to go to.
+     */
+    private Vector3i findRequiredBuilding(BuildingType buildingType, TaskComponent oreonTaskComponent) {
+        List<EntityRef> areas = oreonHolding.assignedAreas;
+        int index = 0;
+        for (EntityRef area : areas) {
+            AssignedAreaComponent areaComponent = area.getComponent(AssignedAreaComponent.class);
+
+            if (areaComponent.buildingType.equals(buildingType)) {
+                oreonTaskComponent.taskRegion = areaComponent.assignedRegion;
+                oreonTaskComponent.assignedAreaIndex = index;
+
+                return areaComponent.assignedRegion.min();
+            }
+
+            index++;
+        }
+
+        //could not find required building
```

Comment:
This might be one of the rare examples of a comment not really being needed, since the log entry says the same thing 😁